src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -45,17 +49,29 @@ var _ webhook.Validator = &Cluster{}
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *Cluster) ValidateCreate() error {
 	log.Info("validate create", "name", r.Name)
-
-	// TODO(user): fill in your validation logic upon object creation.
 	return nil
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *Cluster) ValidateUpdate(old runtime.Object) error {
 	log.Info("validate update", "name", r.Name)
+	oldCluster := old.(*Cluster)
```

Comment:
this is safe to be done because of how the code in controller-runtime is written and considering we cannot get any other object here